lib/browser/api/menu-item-roles.js

```diff
@@ -176,6 +239,21 @@ exports.getDefaultAccelerator = (role) => {
   if (roles.hasOwnProperty(role)) return roles[role].accelerator
 }
 
+exports.getDefaultSubmenu = (role) => {
+  if (roles.hasOwnProperty(role)) {
+    let submenu = roles[role].submenu
+
+    // remove empty objects from within the submenu
+    if (Array.isArray(submenu)) {
+      submenu = submenu.filter(function (n) {
+        return n.constructor !== Object || Object.keys(n).length > 0
```

Comment:
What about if instead of empty objects you just had `null` in the data structure, then this check becomes much simpler as `return n == null`